addons/website_sale/controllers/main.py

```diff
@@ -591,20 +591,21 @@ def checkout_form_validate(self, mode, all_form_values, data):
         if data.get("vat") and hasattr(Partner, "check_vat"):
             if country_id:
                 data["vat"] = Partner.fix_eu_vat_number(country_id, data.get("vat"))
-            partner_dummy = Partner.new({
-                'vat': data['vat'],
-                'country_id': country_id or False,
-            })
+            partner_dummy = Partner.new(self._get_vat_validation_fields(data))
             try:
                 partner_dummy.check_vat()
-            except ValidationError:
+            except ValidationError as exception:
                 error["vat"] = 'error'
+                error_message.append(exception.args[0])
 
         if [err for err in error.values() if err == 'missing']:
             error_message.append(_('Some required fields are empty.'))
 
         return error, error_message
 
+    def _get_vat_validation_fields(self, data):
+        return {'vat': data['vat'], 'country_id': (int(data['country_id']) if data.get('country_id') else False)}
```

Comment:
```suggestion
        return {
            'vat': data['vat'], 
            'country_id': int(data['country_id']) if data.get('country_id') else False,
        }
```

- more readable
-  without extra ()
-  with trailing ,  (avoid a diff on this line if we add a new line in the future)